spaceR/spaceR/ViewModels/LaunchpadViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  LaunchpadViewModel.swift
+//  SpaceR
+//
+//  Created by Kieran Woodrow on 2022/03/21.
+//
+
+import Foundation
+
+class LaunchpadViewModel {
+    
+    private var allLaunchpads: [Launchpads]
+    
+    init() {
+        self.allLaunchpads = []
+    }
+    
+    func setAllLaunchpads(launchpads: [Launchpads]) {
+        allLaunchpads = launchpads
+    }
+    
+    func getAllRockets() -> [Launchpads] {
+        return allLaunchpads
+    }
+    
+    func getLaunchpadCount() -> Int {
+        return allLaunchpads.count
+    }
+    
+    func getLaunchpadName(index: Int) -> String {
+        return allLaunchpads[index].lauchpadName ?? ""
+    }
+    
+    func getLaunchpadImage(index: Int) -> String {
+        var launchpadImage = ""
+        if let safeImage = allLaunchpads[index].images?.largeImage.randomElement() {
+            launchpadImage = safeImage ?? ""
+        }
+        return launchpadImage
+    }
+    
+    func getLaunchpadTitle(index: Int) -> String {
```

Comment:
Consider returning a full object for these functions below 😄

Reply:
Hi @kieranwoodrow so something to think about is how many times we are doing the same thing, so for example if you wanted to show this information on a cell for example you would have to do something like this: 


`viewModel.getLaunchpadTitle(index: indexPath.item)`
`viewModel.getLaunchpadLocation(index:  indexPath.item)`
`viewModel.getLaunchpadRegion(index: indexPath.item)`
`viewModel.getLaunchpadDetails(index: indexPath.item)`
and so on....

Whereas passing the entire object back keeps your code clean and readable and you would only need to unpack things in your cell.

If you are not using all this info you can remove these functions.

Reply:
Let me know if that makes sense 😄

Reply:
cool, removed some future functions